### PR TITLE
[jest-diff] [jest-get-type] [jest-matcher-utils] Add typings

### PR DESCRIPTION
**Pull request by myabc**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/jest-matcher-utils/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+// Project: http://facebook.github.io/jest/
+// Definitions by: Alex Coles <https://github.com/myabc>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.2
+
+import * as chalk from 'chalk';
+
+export const EXPECTED_COLOR: chalk.ChalkChain;
+export const RECEIVED_COLOR: chalk.ChalkChain;
+// export const EXPECTED_BG: chalk.ChalkChain; // removed in b430e51a
```
Comment (myabc) [inline types/jest-matcher-utils/index.d.ts]: N.B. @pedrottimark removed these constants in a commit on `master` (yet to be released) https://github.com/facebook/jest/commit/b430e51a34b389c0f43dfdcf05fab5220a7b5677#diff-98e61ded296e35d3a4fcae207101edeb - so I guess they can be considered de=facto "deprecated"

File: types/jest-matcher-utils/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+// export const RECEIVED_BG: chalk.ChalkChain; // removed in b430e51a
+export const SUGGEST_TO_EQUAL: string;
+
+export function stringify(object: any, maxDepth?: number): string;
+
+export function highlightTrailingWhitespace(
+    text: string,
+    bgColor: chalk.ChalkChain // removed in b430e51a
```
Comment (myabc) [inline types/jest-matcher-utils/index.d.ts]: Also removed in https://github.com/facebook/jest/commit/b430e51a34b389c0f43dfdcf05fab5220a7b5677#diff-98e61ded296e35d3a4fcae207101edeb (PR https://github.com/facebook/jest/pull/4578). I'm not 100% sure if this is the right way to handle this.

File: types/jest-matcher-utils/index.d.ts (hunk 1 above)
Comment (plantain-00) [reply types/jest-matcher-utils/index.d.ts]: If it is not released, it shouldn't be removed now.

Comment (myabc) [reply types/jest-matcher-utils/index.d.ts]: @plantain-00 ok, thanks.

**Review by plantain-00 — APPROVED** [review]

File: types/jest-matcher-utils/index.d.ts (hunk 2 above)
Comment (weswigham) [reply types/jest-matcher-utils/index.d.ts]: You can leave it in and always open a PR with an update once a new version without it is published - then there should be a published `@types` version both with and without it, so people on either version can get what they need.

**Pull request merged**